src/drive/ducks/services/components/SuggestionProvider.jsx

```diff
@@ -77,4 +79,34 @@ class SuggestionProvider extends React.Component {
   }
 }
 
+function getIconUrl(mimetype) {
+  const iconsContext = require.context(
+    '../../../assets/icons/',
+    false,
+    /icon-type-.*.svg$/
+  )
+  const icons = iconsContext.keys().reduce(
+    (acc, item) => ({
+      ...acc,
+      [item.replace(/\.\/icon-type-(.*)\.svg/, '$1')]: iconsContext(item)
+    }),
+    {}
+  )
+
```

Comment:
I am not a fan of this construct. IMHO it is nor very legible, nor performant (you are creating one object and trashing it at every iteration of the reduce). Even if performance is not an issue here, I would prefer if we would not create pressure on the GC during loading of the application.

lodash has a `fromPairs` function that is useful in this case. It is easy to recode it if you want to avoid the dependency on lodash.

```js
const fromPairs = (arr, pairs) => {
  const out = {}
  pairs.forEach(([k, v]) => out[k] = v)
  return out
}

const icons = fromPairs(iconsContext.keys().map(item =>
  [item.replace(/\.\/icon-type-(.*)\.svg/, '$1'), iconsContext(item)])
```

Reply:
You are very right, I did not see it with _the nose on the steering_.
I first wrote: 
```js
const icons = iconsContext.keys().reduce((acc, item) => {
  acc[item.replace(/\.\/icon-type-(.*)\.svg/, '$1')] = iconsContext(item)
  return acc
}, {})
```

That seems enough to avoid what you remark.

Reply:
Yes, I am ok with this 👍